### PR TITLE
Add Live Channels Demo

### PR DESCRIPTION
**Pull request by RikHeijdens**
Added a demo that demonstrates how the JW Live V2 API can be leveraged to replace a VOD asset that plays on a page in a loop with a livestream, as soon as the livestreaming Channel becomes active.

### CODE REVIEW

File: demos/live-channels/demo.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
+const selectButton = document.getElementById('select-channel');
+const channelInput = document.getElementById('channel-input');
+const statusMessage = document.getElementById('status-message');
+
+/** The player config to use in order to initialize the player */
+const VOD_CONFIG = {
+  playlist: 'https://cdn.jwplayer.com/v2/media/8L4m9FJB',
+  // Repeat the VOD indefinitely while we wait for the livestream to become available.
```
Comment (kcorneli201) [inline demos/live-channels/demo.js:8]: // Repeat the VOD indefinitely while we wait for the live event stream to become available.

File: demos/live-channels/demo.js [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+  }
+
+  statusMessage.textContent = `Waiting for live channel ${channelId} to become active.`;
+
+  // Start the update loop.
+  checkChannelStatus();
+});
+
+/**
+ * Periodically checks whether the specified livestream channel is available, and if it is, configures the player
```
Comment (kcorneli201) [inline demos/live-channels/demo.js:55]: Periodically checks whether the specified live event stream is available...etc.

File: demos/live-channels/demo.js [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+ */
+function checkChannelStatus() {
+  if (!intervalId) {
+    // Make sure to execute this method every UPDATE_FREQUENCY milliseconds.
+    intervalId = setInterval(checkChannelStatus, UPDATE_FREQUENCY);
+  }
+  getChannelStatus(channelId).then((channelStatus) => {
+    console.log(`Received channel status: %O`, channelStatus);
+    if (channelStatus['status'] === 'active') {
+      // Determine the id of the active event based on the returned status.
```
Comment (kcorneli201) [inline demos/live-channels/demo.js:66]: // Determine the ID....etc.

File: demos/live-channels/demo.js [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+      const eventId = channelStatus['current_event'];
+      // Attempt to configure the player in order to start livestream playback.
```
Comment (kcorneli201) [inline demos/live-channels/demo.js:68]: // Attempt to configure the player in order to start live event stream playback.

File: demos/live-channels/demo.js [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+    }
+  }, (error) => {
+    statusMessage.textContent = `Unable to fetch the channel status for ${channelId}: ${error}`;
+    // If we fail to retrieve the channel status, then give up.
+    clearInterval(intervalId);
+  });
+}
+
+/**
+ * (Re-)configures the active playerInstance to play the livestream identified by eventId.
```
Comment (kcorneli201) [inline demos/live-channels/demo.js:82]: * (Re-)configures the active playerInstance to play the live event stream identified by eventId.

File: demos/live-channels/demo.js [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+ */
+async function configurePlayer(eventId) {
+  // There may be a slight delay between the livestream becoming available, and its playlist to become available.
```
Comment (kcorneli201) [inline demos/live-channels/demo.js:85]: // There may be a slight delay between the live event stream becoming available, and its playlist to become available.

File: demos/live-channels/demo.js [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+  // Therefore, we first attempt to fetch the playlist for the new live event, as soon as we have successfully fetched
+  // a playlist, we will load it on the player and start playback of the livestream.
```
Comment (kcorneli201) [inline demos/live-channels/demo.js:87]: // a playlist, we will load it on the player and start playback of the live event stream.

File: demos/live-channels/demo.js [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+    .then((response) => {
+      if (!response.ok) {
+        throw new Error(`Unable to fetch ${url}: ${response.statusText}`);
+      }
+      return response.json();
+    });
+}
+
+/**
+ * Fetches the current status of a JW Live 2.0 channel.
```
Comment (kcorneli201) [inline demos/live-channels/demo.js:131]: * Fetches the current status of a live event stream within a Live Channel.

File: demos/live-channels/demo.js [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+ * Returns a promise that will yield the status for a particular channel.
+ *
+ * @param channelId The channel to fetch the status for.
+ */
+function getChannelStatus(channelId) {
+  return fetchJSON(`https://cdn.jwplayer.com/live/channels/${channelId}.json`);
+}
+
+/**
+ * Fetches a JW Platform playlist for a particular media item.
```
Comment (kcorneli201) [inline demos/live-channels/demo.js:141]: * Fetches a playlist for a particular media item.

File: demos/live-channels/index.html [hunk 10]
```diff
@@ -0,0 +1,32 @@
+<div id="player"></div>
+
+<form id="demo-form" autocomplete="off">
+	<input class="ds-textinput" type="text" id="channel-input" placeholder="Enter a channel id" />
```
Comment (kcorneli201) [inline demos/live-channels/index.html:4]: Enter a channel ID

File: demos/live-channels/index.html [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+	<button id="select-channel" class="ds-button--primary">Select Channel</button>
+  <span id="status-message">Waiting for a Live Channel to be specified.</span>
+</form>
+
+<p>
+	Utilize live streaming with JW Platform and JW Player by using JW Live as
+	your live streaming provider.
```
Comment (kcorneli201) [inline demos/live-channels/index.html:11]: Utilize JW Player's Live Channels as your live event streaming solution.

File: demos/live-channels/index.html [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+</p>
+<p>
+	This demo demonstrates how to place a player on your page which will play a
+	VOD asset in a loop until a JW Live Channel becomes available.
```
Comment (kcorneli201) [inline demos/live-channels/index.html:15]: VOD asset in a loop until a live event stream becomes available.

File: demos/live-channels/index.html [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+</p>
+<h2>Usage</h2>
+<ol>
+	<li>Enter the ID of your <a href="https://dashboard.jwplayer.com/#/content/live/channels">Live Channel</a> and click "Select Channel".</li>
+	<li>Start broadcasting to your channel.</li>
+</ol>
+<p>
+	After hitting "Select Channel" the code on this page will automatically check for availability
+	of the channel in the background and replace the VOD asset in the player with the livestream
```
Comment (kcorneli201) [inline demos/live-channels/index.html:24]: of a live event stream in the Live Channel in the background and replace the VOD asset in the player with the live event stream

File: demos/live-channels/index.html [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+	associated with the channel.
+</p>
+<p>
+	<a href="https://support.jwplayer.com/articles/how-to-setup-dvr-live-streaming">Visit our support site</a>
+	for a tutorial on live streaming with JW Platform and JW Player, or
```
Comment (kcorneli201) [inline demos/live-channels/index.html:29]: for a tutorial on streaming a live event with JW Player, or

File: demos/live-channels/index.html [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+	<a href="https://www.jwplayer.com/contact-us/?utm_source=developer&utm_medium=CTA&utm_campaign=player-demos">contact us</a>
+	for more information on JW Live Channels.
```
Comment (kcorneli201) [inline demos/live-channels/index.html:31]: for more information on Live Channels.

File: demos/live-channels/demo.js (hunk 9 above)
Comment (RikHeijdens) [reply demos/live-channels/demo.js:141]: I have intentionally used the JW Platform playlist wording here in order to disambiguate between HTTP Live Streaming playlists and JW Platform playlists.

File: demos/live-channels/demo.js (hunk 2 above)
Comment (RikHeijdens) [reply demos/live-channels/demo.js:55]: We actually check the status of the channel here, not the event.

**Comment by RikHeijdens** [comment]
Thanks for reviewing @kcorneli201, I've addressed most of your comments in order to ensure that our language is consistent between our documentation and demos.

I also left some occurrences of the term `livestream` because some of these comments apply to HTTP Live Streams (as in [RFC-8216](https://tools.ietf.org/html/rfc8216)) in general and not necessarily to JW Player Live Streams.

**Pull request merged**